Merge Meeds-io/MIPs#63

boubaker opened the pull request.

boubaker committed:
feat: Rework General Settings Page to diffferent sections - MEED-1652 - Meeds-io/MIPs#63

boubaker committed:
feat: Add Transition when expanding main setting section - MEED-1652 - Meeds-io/MIPs#63

boubaker committed:
feat: Allow to store Registration Settings from UI - MEED-1645 - Meeds-io/MIPs#63

This change will allow to store Security Settings of Registered users
Access Type and default permissions.

boubaker committed:
feat: Allow to choose default spaces when registering a user - MEED-2425 - Meeds-io/MIPs#63

This change will introduce a UI to allow choosing default spaces that
will be added to a newly registered user.

boubaker committed:
feat: Add Contextual Help to User Registration settings - MEED-2426 - Meeds-io/MIPs#63

This change will add a dedicated drawer to give more information about
specific items and terms for end user.

boubaker committed:
feat: Add direct Access to Main Settings Tab - MEED-2429 - Meeds-io/MIPs#63

rdenarie commented on component/oauth-auth/src/main/java/io/meeds/oauth/service/impl/OAuthRegistrationServiceImpl.java:
```diff
@@ -92,7 +98,8 @@ public OAuthRegistrationServiceImpl(PortalContainer container,
 
   @Override
   public boolean isRegistrationOnFly(OAuthProviderType<? extends AccessTokenContext> oauthProviderType) {
-    return registerOnFly.contains(oauthProviderType.getKey());
```
In customer contexts, we need registration on the fly even if registrationType is restricted : 

Functionnal need : 
exo is configured with openid or oauth, but not connected to the AD/ldap. Users are not known in the platform until they log the first time.
In this case, the registration type is restricted AND registerOnTheFly contains OPENID.

With this modification, the user cannot log in and create an account, but is it a valid behaviour

boubaker replied:
ok, this was discussed in a Task as functional behavior need. @srenault-meeds , we may missed a special use case, where it's restricted but Open on some Registration providers. WDYT ?

srenault-meeds replied:
Hi, can we discuss it please @rdenarie? 
Indeed, I need to understand the case to make sure we adapt or not.

srenault-meeds replied:
After discussion with both parties, here is what we will do:
- enable registeronthefly option even when the platform is restricted
- no ability to change this by UI, only by configuration
- so signup by email won't be possible but with oAuth provider enabled and configured with proper clientsecret
Thanks for your understanding

boubaker replied:
ok done here 119557e9ab18e7af7e0fad2054b2c44d554c065e
Thanks for both.

boubaker committed:
feat: Add Message Warning when closing Main Setting Tab - MEED-2465 - Meeds-io/MIPs#63 (#2810)

This change will introduce a confirm step before closing the Main
Setting Tab when changed to ensure that user doesn't loose his changes.

boubaker committed:
feat: Allow to Register on Fly when Restricted - MEED-2470 - Meeds-io/MIPs#63
Prior to this change, the restricted Registration Type wasn't allowing to register on the fly using OAuth Providers. This change will decouple the OAuth behavior from Restricted Registration Type.
